File: solutions/julia/tisbury-treasure-hunt/1/tisbury-treasure-hunt.jl
function get_coordinate(line)
    line[2]
end

function convert_coordinate(coordinate)
    Tuple(coordinate)
end

function compare_records(azara_record, rui_record)
    convert_coordinate(azara_record[2]) == rui_record[2]
end

function create_record(azara_record, rui_record)
    if compare_records(azara_record, rui_record)
        (azara_record[2], rui_record[1], rui_record[3], azara_record[1])
    else
        ()
    end
end

File: solutions/fsharp/tisbury-treasure-hunt/1/TisburyTreasureHunt.fs
module TisburyTreasureHunt

let getCoordinate (line: string * string) : string = snd line

let convertCoordinate (coordinate: string) : int * char =
    (int coordinate[0] - int '0', coordinate[1])

let compareRecords (azarasData: string * string) (ruisData: string * (int * char) * string) : bool =
    let (a, (b, c), d) = ruisData in snd azarasData = b.ToString() + c.ToString()

let createRecord
    (azarasData: string * string)
    (ruisData: string * (int * char) * string)
    : (string * string * string * string) =
    if compareRecords azarasData ruisData then
        let (a, b) = azarasData in
        let (c, (d, e), f) = ruisData in
        (b, c, f, a)
    else
        ("", "", "", "")
